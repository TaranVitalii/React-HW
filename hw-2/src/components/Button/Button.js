import React from "react";

export default props => {
  function clickHandler(e) {
    props.onClick(props.data);
  }
  return (
    <button className="Button" onClick={clickHandler}>
      Submit
    </button>
  );
};
