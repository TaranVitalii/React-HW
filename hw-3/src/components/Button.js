import React from "react";

export default function Button({ onClickHandler, figure, styleButton }) {
  return (
    <button
      className={`button${figure}`}
      onClick={onClickHandler}
      style={styleButton}
    >
      {figure}
    </button>
  );
}
