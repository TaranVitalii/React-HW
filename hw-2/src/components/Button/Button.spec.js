import React from "react";
import Button from "./Button";
import { shallow } from "enzyme";

describe("<Button/>", () => {
  it("should render a Button", () => {
    const wrapper = shallow(<Button />);

    expect(wrapper).toMatchSnapshot();
  });
});
