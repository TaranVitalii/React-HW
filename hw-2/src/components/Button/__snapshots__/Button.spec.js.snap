// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render a Button 1`] = `ShallowWrapper {}`;

exports[`<InputForm/> 1st test 1`] = `ShallowWrapper {}`;
